core/templates/components/state-editor/state-editor.directive.ts

```diff
@@ -121,6 +125,27 @@ angular.module('oppia').directive('stateEditor', [
             $scope.conceptCardIsShown = !$scope.conceptCardIsShown;
           };
 
+          var questionSuggestionReview = angular.element(
+            document.querySelector('.oppia-question-suggestion-review'));
+          questionSuggestionReview.on('scroll', function() {
+            var tutorialStateContent = document.getElementById(
+              'tutorialStateContent');
+            var cardAvatarOnSticky = document.querySelector(
+              '.oppia-editor-card-avatar');
+            if (questionSuggestionReview.scrollTop() >
+             STICKY_HEADER_SCROLL_TOP) {
+              tutorialStateContent.classList.remove(
+                'state-content-editor-parent-container');
+              cardAvatarOnSticky.classList.add(
+                'oppia-editor-card-avatar-sticky');
+            } else {
+              tutorialStateContent.classList.add(
+                'state-content-editor-parent-container');
+              cardAvatarOnSticky.classList.remove(
+                'oppia-editor-card-avatar-sticky');
+            }
+          });
```

Comment:
Why can't we just toggle a boolean here and use `ng-class` like in https://github.com/oppia/oppia/pull/11738/files?

Reply:
I had used the ng-class and implemented similar to https://github.com/oppia/oppia/pull/11738/files referred by you, writing a function which toggled the Boolean. But @kevintab95 expressed concerns about "...triggering a `$applyAsync` everytime the scroll event is triggered." So I replaced the function with the current one that does not call `$applyAsync()`.

Reply:
Well, you could trigger an `$applyAsync` only when there is an actual change in the value used in the template, right? In the previous scenario, you would trigger a digest loop irrespective of whether there is an actual change in the `isSticky` value or not -- this is not efficient.

Reply:
Not clear what you are conveying here. There are two codes written on similar lines. One is merged around Mar this year (you are one of the reviewers there), and the other code is this one. What is the difference that causes one to be inefficient and the other efficient?

![image](https://user-images.githubusercontent.com/68392962/116699275-0155c280-a9e3-11eb-9bb3-c5c7681882e5.png)

@sagangwee - There are two codes, one with ng-class/Boolean and another currently reflecting in the file changes. I am reverting to you to conclude on this.

Reply:
Has this been addressed?

Reply:
Can you tell me which part of my explanation is unclear and I'll try to rephrase it?

Reply:
Thanks for catching the fact that a similar code exists. It may be that it was missed in review. Nevertheless, it is worth revisiting. Also, looking at the PR in question https://github.com/oppia/oppia/pull/11738/files#diff-76d42d4e9c4341a0b15dc4ac96bdf2a88ba95477a0dbeb34ece4bc45a4b34768R110-R119, it doesn't look like I had reviewed it (I am not a codeowner in any of the changed files there).